utils/nsdialer_linux.go

```diff
@@ -0,0 +1,45 @@
+// +build linux
+
+package utils
+
+import (
+	"context"
+	"net"
+
+	"github.com/pkg/errors"
+	"go.uber.org/multierr"
+
+	"github.com/vishvananda/netns"
+)
+
+func (n *NsDialer) DialContext(ctx context.Context, network, address string) (net.Conn, error) {
+	origNs, err := netns.Get()
```

Comment:
Don't we need to lock our runtime thread here?

Reply:
Yup. Thanks.